src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Commit.cs

```diff
@@ -95,23 +94,39 @@ private void Commit(PresentationItem item, Model model, char? commitChar, Cancel
                         IgnoreTrimWhiteSpace = EditOptions.DefaultMinimalChange.DifferenceOptions.IgnoreTrimWhiteSpace
                     });
 
-                    // edit subject buffer (not view) because text changes are in terms of current document.
-                    using (var textEdit = this.SubjectBuffer.CreateEdit(editOptions, reiteratedVersionNumber: null, editTag: null))
+                    var textChangesInCurrentViewSnapshot = new List<TextChange>();
+
+                    foreach (var triggerSnapshotTextChange in textChangesAgainstTriggerSnapshot)
                     {
-                        for (int iChange = 0; iChange < textChanges.Length; iChange++)
+                        var currentSpanInBuffer = new SnapshotSpan(
+                            model.TriggerSnapshot,
+                            triggerSnapshotTextChange.Span.ToSpan());
+
+                        // Try mapping the item span against the trigger snapshot to the ViewBuffer's trigger snapshot.
+                        // Then map that forward to the ViewBuffer's current snapshot.
+                        var originalSpanInView = model.GetViewBufferSpan(triggerSnapshotTextChange.Span).TextSpan.ToSnapshotSpan(model.TriggerSnapshot);
+                        var translatedOriginalSpanInView = originalSpanInView.TranslateTo(TextView.TextBuffer.CurrentSnapshot, SpanTrackingMode.EdgeInclusive);
+
+                        textChangesInCurrentViewSnapshot.Add(new TextChange(translatedOriginalSpanInView.Span.ToTextSpan(), triggerSnapshotTextChange.NewText));
+                    }
+
+                    // Note: we currently create the edit on the textview's buffer.  This is 
+                    // necessary as our own 
+                    using (var textEdit = TextView.TextBuffer.CreateEdit(editOptions, reiteratedVersionNumber: null, editTag: null))
+                    {
+                        for (int iChange = 0; iChange < textChangesAgainstTriggerSnapshot.Length; iChange++)
```

Comment:
i don't like that we're using two different arrays for length and for indexing.  can you only reference one of the arrays here?